### ISSUE TITLE
Configure travis CI to push to ross-website repository

### ISSUE THREAD

**Issue by raphaeltimbo** [ross-rotordynamics/ross#266]
To update our website we need to push the updated generated docs to the ross-website repository.
We can do this automatically by setting things on Travis CI.
The idea is that everytime we merge some code we are going to have the website updated.
More instructions on this are given [here](https://docs.travis-ci.com/user/deployment/pages/).

- [x] Create script to build documentation and deploy to ross-website repository;
- [x] Generate SSH keys that will be used to deploy from travis and add the encrypted key (deploy_key.enc) to the github repository;
- [x] Edit .travis.yml to add the encryption label;

**Comment by raphaeltimbo** [comment]
> More instructions on this are given [here](https://docs.travis-ci.com/user/deployment/pages/).

The approach used on this link has the following problem described [here](https://gist.github.com/domenic/ec8b0fc8ab45f39403dd):
_NOTE: an earlier version of this guide recommended generating a GitHub personal access token and encrypting that. Although this is simpler, it is not a good idea in general, since it means any of your repository's collaborators would be able to edit the Travis build script to email them your access token, thus giving them access to all your repositories. The repository-specific deploy key approach is safer._

So I will try to take this other approach and generate SSH keys to deploy with travis.

**Comment by raphaeltimbo** [comment]
This can be closed now.
Just to document the process in case we need to change this latter, I did not implement this in a PR because I needed to push directly to master to trigger the deployment of the docs to the ross-website repository.
The complete process goes from commit bec3d18042c14e70016a8a034607429042f1df33 to commit 45046c4f8d80a9577a1f5853009551ae9355cf06.

Some point worth mentioning:
- When generating the deploy_key.enc, the travis encrypt command needs to be run on the ross repository folder and not on the ross-website folder. The environment variables key and iv from this command: `openssl aes-256-cbc -K $encrypted_904ffbd6830c_key -iv $encrypted_904ffbd6830c_iv -in deploy_key.enc -out deploy_key -d` only exist on a travis build started from the ross repository.
- If we add the ssh key before cloning the repository we are able to make the clone already using the ssh protocol instead of https;
- Some libraries used to build the documentation were added to the dev extra requirements;
- Pandoc had to be added to the before_install travis step, since it is needed to convert the jupyter notebooks to html;
